### DISCUSSION TITLE
How to use Web3Modal V2 with Ethers js

### DISCUSSION THREAD

**Discussion by cryptobredda** [WalletConnect/walletconnect-monorepo#2671] [Developer Support] [answered]
Hi,

I would like to see a tutorail on how I would be able to use the new Web3Modal V2 with ethers js.

I am not a big fan of wagami and I would prefer using ethers js instead and creating my own hooks than using the prebuilt hooks. I used Web3Modal V1 in this way for a very long time and I am comfortable with that. 

Is there a way where that I can use Web3Modal as the proivder for the ethers js and then use ethers js as normal?

Thank you

**Comment by glitch-txs** [comment]
You can use ethereum-provider package but it doesn't come with browser wallets on the modal

**Reply by cryptobredda** [reply]
So this wont give me all the wallet but just the injected provider ones?

**Reply by glitch-txs** [reply]
You'll get the modal with the QR code but no browser wallets like MetaMask

**Reply by cryptobredda** [reply]
I see, thank you for the information.

After further research, I found that wagmi has something called Ethers js Adapter which allows the usage of the classic ethers js.

Now I havent used this function before and I am not sure how to implement it. Do you have any experience using that?

**Reply by glitch-txs** [reply]
Where did you find that? I'm not aware of that

**Reply by cryptobredda** [reply]
In Wagmi docs 

At the bottom you should see Ethers Adapter section. I had a little play with but it seems a little weird. Like I can access the provider using ethers but not all functions are accessable. for example,

Getting the chain id from , Works well.

`const { chainId } = await provider.getNetwork()`

but trying to get the signer from the provider doesnt work.  I am aware there is a hook for the signer as well but again its little weird as well.
the docs doesnt explain it well unfortunatly thus I am looking for tutorial or help from an experienced users.


![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/36314021/486b8bc7-b200-44a4-b256-138b79a606cf)

**Reply by glitch-txs** [reply]
Ah I see, yes because wagmi has a walletClient as signer

**Reply by cryptobredda** [reply]
Yeah its a little weird how to convert the two, wagmi use veim as their eth interface.

what would you then suggest for me to do mate? shall I take this with wagmi then?

**Reply by glitch-txs** [reply, marked as answer]
I would give it a try to viem, but it's up to you, otherwise you can try handle this yourself and go with ethereum-provider which is also great choice, but you'll need to handle more things, like the account state, etc
